hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -483,417 +493,444 @@ private int getWhenNoSSLDetected() {
-     * execution of the algorithm.
+     * The partial member disconnection resolution mechanism uses a graph algorithm
+     * that finds a max-clique in non-polynomial time. Since it could take a lot
+     * of time to find a max-clique in a large graph, i.e, in a large cluster
+     * with lots of random network disconnections, we use a timeout mechanism
+     * to stop execution of the algorithm.
      */
     public static final HazelcastProperty PARTIAL_MEMBER_DISCONNECTION_RESOLUTION_ALGORITHM_TIMEOUT_SECONDS
             = new HazelcastProperty("hazelcast.partial.member.disconnection.resolution.algorithm.timeout.seconds", 5);
 
     /**
      * Heartbeat failure detector type. Available options are:
      * <ul>
-     * <li><code>deadline</code>:  A deadline based failure detector uses an absolute timeout
+     * <li>{@code deadline}:  A deadline based failure detector uses an absolute timeout
      * for missing/lost heartbeats. After timeout member is considered as dead/unavailable.
      * </li>
-     * <li><code>phi-accrual</code>: Implementation of 'The Phi Accrual Failure Detector' by Hayashibara et al.
-     * as defined in their paper. Phi Accrual Failure Detector is adaptive to network/environment conditions,
-     * that's why a lower {@link #MAX_NO_HEARTBEAT_SECONDS} (for example 10 or 15 seconds) can be used to provide
-     * faster detection of unavailable members.
+     * <li>{@code phi-accrual}: Implementation of 'The Phi Accrual Failure Detector'
+     * by Hayashibara et al. as defined in their paper. Phi Accrual Failure Detector
+     * is adaptive to network/environment conditions, that's why a lower
+     * {@link #MAX_NO_HEARTBEAT_SECONDS} (for example 10 or 15 seconds) can be
+     * used to provide faster detection of unavailable members.
      * </li>
      * </ul>
-     *
+     * <p>
      * Default failure detector is <code>deadline</code>.
      */
     public static final HazelcastProperty HEARTBEAT_FAILURE_DETECTOR_TYPE
             = new HazelcastProperty("hazelcast.heartbeat.failuredetector.type", ClusterFailureDetectorType.DEADLINE.toString());
 
     /**
-     * The interval at which the master sends the member lists are sent to other non-master members
+     * The interval at which the master sends the member lists are sent to other
+     * non-master members.
      */
     public static final HazelcastProperty MEMBER_LIST_PUBLISH_INTERVAL_SECONDS
             = new HazelcastProperty("hazelcast.member.list.publish.interval.seconds", 60, SECONDS);
 
+    /**
+     * Time after which the member assumes the client is dead and closes its
+     * connections to the client.
+     */
     public static final HazelcastProperty CLIENT_HEARTBEAT_TIMEOUT_SECONDS
             = new HazelcastProperty("hazelcast.client.max.no.heartbeat.seconds", 300, SECONDS);
 
     public static final HazelcastProperty CLUSTER_SHUTDOWN_TIMEOUT_SECONDS
```

Comment:
Could you also add the JavaDoc?

Reply:
Added. I used my power of reasoning and tried to write a nice, fresh javadoc :) Maybe I can add the same now to the reference manual 🤔